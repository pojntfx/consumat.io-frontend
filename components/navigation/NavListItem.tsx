import Link from "next/link";
import styles from "../../styles/Navbar.module.css";

type NavListItemProps = {
  title: string;
  href: string;
  isActive: boolean;
  isMobile?: boolean;
  setMobileNavbarVisibility?: (boolean) => void;
};

const NavListItem = ({
  title,
  href,
  isActive,
  isMobile = false,
  setMobileNavbarVisibility = null,
}: NavListItemProps) => {
  return (
    <li
      className={`${styles.navigationLink} ${
        isMobile ? styles.navigationItemMobile : styles.navigationItem
      } ${isActive && styles.active}`}
    >
      <Link href={href}>
        <a
          onClick={
            setMobileNavbarVisibility &&
            (() => setMobileNavbarVisibility(false))
          }
        >
          {title}
        </a>
      </Link>
    </li>
  );
};

export default NavListItem;
