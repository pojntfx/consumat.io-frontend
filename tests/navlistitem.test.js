import React from "react";
import renderer from "react-test-renderer";
import NavListItem from "../components/navigation/NavListItem";

it("renders correctly", () => {
  const tree = renderer.create(<NavListItem href="/" />).toJSON();
  expect(tree).toMatchSnapshot();
});
