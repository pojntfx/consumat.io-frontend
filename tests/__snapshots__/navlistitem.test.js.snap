// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li>
  <a
    className="navigation-link link navigationLink dark:hover:border-white dark:hover:text-white undefined"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  />
</li>
`;
